packages/svelte/src/internal/client/dom/elements/class.js

```diff
@@ -31,7 +71,10 @@ export function class_name(dom, value) {
 		prev_class_name !== next_class_name ||
 		(hydrating && dom.className !== next_class_name)
 	) {
-		if (next_class_name === '') {
+		// Removing the attribute when the value is only an empty string causes
+		// peformance issues vs simply making the className an empty string. So
+		// we should only remove the class if the the value is nullish.
+		if (value === null) {
```

Comment:
nullish -> also undefined?
```suggestion
		if (value == null) {
```